### PR TITLE
feat: add a safe storage registry

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?

introduce a registry storing key/values in encrypted and then encoded in base64 values

### Screenshot / video of UI

N/A

### What issues does this PR fix or reference?

part 1 of https://github.com/containers/podman-desktop/issues/6044

### How to test this PR?

unit tests added

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

File: packages/main/src/plugin/safe-storage/safe-storage-registry.ts [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+export class SafeStorageRegistry {
+  readonly #directories: Directories;
+
+  #extensionStorage: SafeStorage | undefined;
+
+  constructor(directories: Directories) {
+    this.#directories = directories;
+  }
+
+  protected getSafeStoragePath(): string {
```
Comment (jeffmaury) [inline packages/main/src/plugin/safe-storage/safe-storage-registry.ts]: Name is confusing something like getSafeStorageConfigPath

**Review by jeffmaury — APPROVED** [review]
LGTM. Wondering if having a specific separate storage is necessary, could'nt be part of the configuration API ?

**Comment by benoitf** [comment]
the problem is that if you store that in the configuration registry, you'll be able to see the entries in the UI (in encrypted format)

File: packages/main/src/plugin/safe-storage/safe-storage-registry.ts (hunk 1 above)
Comment (benoitf) [reply packages/main/src/plugin/safe-storage/safe-storage-registry.ts]: ok but it's not the configuration path it's where the data are stored

Comment (benoitf) [reply packages/main/src/plugin/safe-storage/safe-storage-registry.ts]: `getSafeStorageDataPath` ?

Comment (benoitf) [reply packages/main/src/plugin/safe-storage/safe-storage-registry.ts]: fixed

**Comment by benoitf** [comment]
rebased

**Comment by benoitf** [comment]
> Base64 is solely an encoding mechanism, not a form of encryption. I can't find the follow up issues/explanation, is this temporary ?

@axel7083 so it's encoded in base64 just to be sure that we can easily read/write files / make it compliant with utf-8 etc

but before the encoding it's encrypted

**Comment by axel7083** [comment]
> > Base64 is solely an encoding mechanism, not a form of encryption. I can't find the follow up issues/explanation, is this temporary ?
> 
> @axel7083 so it's encoded in base64 just to be sure that we can easily read/write files / make it compliant with utf-8 etc
> 
> but before the encoding it's encrypted

I deleted my comment after reading properly the code, yes reading your pull request changes made me realize the description was not what I thought

**Review by axel7083 — APPROVED** [review]

**Comment by benoitf** [comment]
I've updated the description to make it clearer

**Pull request merged**